crates/re_viewer/src/ui/view_spatial/scene/scene_part/images.rs

```diff
@@ -277,24 +278,53 @@ impl ImagesPart {
             return Err(format!("Couldn't fetch pinhole extrinsics at {pinhole_ent_path:?}"));
         };
 
-        // TODO(cmc): automagically convert as needed for non-natively supported datatypes?
-        let data = match &tensor.data {
-            // NOTE: Shallow clone if feature `arrow` is enabled, full alloc + memcpy otherwise.
-            TensorData::U16(data) => DepthCloudDepthData::U16(data.clone()),
-            TensorData::F32(data) => DepthCloudDepthData::F32(data.clone()),
-            _ => {
-                return Err(format!(
-                    "Tensor datatype {} is not supported for backprojection",
-                    tensor.dtype()
-                ));
-            }
+        let (height, width) = (tensor.shape()[0].size, tensor.shape()[1].size);
+        let dimensions = glam::UVec2::new(width as _, height as _);
+
+        // Ideally, we'd use the same key as for displaying the texture, but we might make other compromises regarding formats etc.!
+        // So to not couple this, we use a different key here
+        let depth_texture = {
+            let texture_key = egui::util::hash((tensor.id(), "depth_cloud"));
+            let mut data_f32 = Vec::new();
+            ctx.render_ctx.texture_manager_2d.get_or_create_with(
+                texture_key,
+                &mut ctx.render_ctx.gpu_resources.textures,
+                || {
+                    // TODO(andreas/cmc): Ideally we'd upload the u16 data as-is.
+                    // However, R16Unorm is behind a feature flag and Depth16Unorm doesn't work on WebGL (and is awkward as this is a depth buffer format!).
+                    let data = match &tensor.data {
+                        TensorData::U16(data) => {
+                            data_f32.extend(
+                                data.as_slice().iter().map(|d| *d as f32 / u16::MAX as f32),
```

Comment:
```suggestion
                                data.as_slice().iter().map(|d| *d as f32),
```

and remove the multiplier by `65535.0` below when creating `world_depth_from_data_depth`